hbase-server/src/main/java/org/apache/hadoop/hbase/procedure2/store/region/RegionProcedureStore.java

```diff
@@ -183,51 +106,14 @@ public void start(int numThreads) throws IOException {
     this.numThreads = numThreads;
   }
 
-  private void shutdownWAL() {
-    if (walFactory != null) {
-      try {
-        walFactory.shutdown();
-      } catch (IOException e) {
-        LOG.warn("Failed to shutdown WAL", e);
-      }
-    }
-  }
-
-  private void closeRegion(boolean abort) {
-    if (region != null) {
-      try {
-        region.close(abort);
-      } catch (IOException e) {
-        LOG.warn("Failed to close region", e);
-      }
-    }
-
-  }
-
   @Override
   public void stop(boolean abort) {
     if (!setRunning(false)) {
       return;
     }
     LOG.info("Stopping the Region Procedure Store, isAbort={}", abort);
-    if (cleaner != null) {
-      cleaner.cancel(abort);
-    }
-    if (flusherAndCompactor != null) {
-      flusherAndCompactor.close();
-    }
-    // if abort, we shutdown wal first to fail the ongoing updates to the region, and then close the
-    // region, otherwise there will be dead lock.
-    if (abort) {
-      shutdownWAL();
-      closeRegion(true);
-    } else {
-      closeRegion(false);
-      shutdownWAL();
-    }
-
-    if (walRoller != null) {
-      walRoller.close();
+    if (localStore != null) {
```

Comment:
Is this the right place to stop this? Should the localStore be made by the Master, owned and shutdown by the Master? It passes it in here to the RPS to use? Master shuts it down. Makes sure it doesn't shut it down before other users are done with it?

Reply:
The ideal answer is, no. It should be initialized by master and also stop by master. You can see the draft PR here
https://github.com/apache/hbase/pull/1746
Where I moved the initialization and stop out from the RegionProcedureStore.

And here, since RegionProcedureStore is the only place where we use the LocalStore, there will be no technical problem to initialize and stop it inside RegionProcedureStore. Moving it out will have big impact on tests, as in most tests we will stop the ProcedureStore and reinitialize it again to test fail recovery. If we move the initialization and stop out, all these tests have to be rewritten as we need to recreate the LocalStore by our own.

So I prefer we do this in another issue, where we do need to store data other than procedure to the LocalStore. For example, in HBASE-24388 I have to do this.

Reply:
Oh, maybe we do not need to restart the local region when testing procedure store restart. Just add some tests to make sure that the local store recovery can work is enough?
Let me have a try.